debian/postinst

```diff
@@ -115,10 +116,28 @@ EOF
     fi
 }
 
+
+mark_reboot_for_fips_pro() {
+    if ! check_service_is_enabled fips; then
```

Comment:
I don't think we can perform this call here. This is because `check_service_is_enabled` will call `ua status` which will then refresh the status cache. For example, during a `do-release-upgrade` , we would update the cache, which will show `esm-infra`  as disabled because of the apt url change and then the `upgrade_lts_contract`  script will not work as expected during reboot.

However, this will not be true if `do-release-upgrader` now correclty updates the apt url on the source list file before upgrading the package.

Reply:
Changed `check_service_is enabled` to  checks the cached status.json value instead of calling out and updating the status.json cache and potentially setting services to disabled as a result. Great catch